### PR TITLE
Update gqlgen and fix the code.

### LINKED ISSUE #1054: Bumping `github.com/99designs/gqlgen` version requires product code changes
There is a new version of the `github.com/99designs/gqlgen` API dependency, but it requires code changes from the product side, so this issue is here to track them.

Here are test execution results with GO build failure: https://github.com/kkovaletp/photoview/actions/runs/10898561504/job/30242036326?pr=147

### PR DESCRIPTION
**Pull request by googollee**
Fix #1054

### CODE REVIEW

File: api/go.mod [hunk 1]
```diff
@@ -1,9 +1,11 @@
 module github.com/photoview/photoview/api
 
-go 1.22.0
+go 1.22.5
```
Comment (googollee) [inline api/go.mod:3]: Note: `go mod tidy` updates this version and adds `toolchain` version, based on all dependencies. It's good to let `go mod tidy` to maintain it.

File: api/go.mod [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
+
+toolchain go1.23.1
```
Comment (kkovaletp) [inline api/go.mod:5]: Is it ok that the `toolchain` version is not synced with the `go` version?

Comment (googollee) [reply api/go.mod:5]: It's OK. `go` means the feature in the code, `toolchain` means compiler version. As we are using `golang-1.23.1` image, it's fine.

Comment (googollee) [reply api/go.mod:5]: It's also managed by `go mod tidy`.

**Comment by googollee** [comment]
I guess [api/graphql/endpoint/graphql_endpoint.go](https://github.com/photoview/photoview/pull/1055/files#annotation_26368893407) is a generated code. I don't find a way to test it easily. Hard to fix `codecov/patch` test.

**Review by kkovaletp — APPROVED** [review]
Approved

**Pull request merged**